### PR TITLE
Coral-Trino: Shade trino-parser

### PR DESCRIPTION
**Pull request by ljfgem**
This PR is to shade `trino-parser` dependency for #205 

Tested it by publishing to local maven and integrating with Trino.

### CODE REVIEW

**Review by shenodaguirguis — APPROVED** [review]

**Review by wmoustafa — COMMENTED** [review]
Could you comment on how it differs from the last shading attempt?

**Comment by ljfgem** [comment]
> Could you comment on how it differs from the last shading attempt?

@wmoustafa Main difference is that it added:
```
apply from: "$rootDir/gradle/java-publication.gradle"
```
like how other modules did

**Review by wmoustafa — APPROVED** [review]

File: shading/coral-trino-parser/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation
+
+plugins {
+  id 'com.github.johnrengelman.shadow'
+}
+
+apply from: "$rootDir/gradle/java-publication.gradle"
+
+dependencies {
+  implementation 'io.trino:trino-parser:355'
```
Comment (wmoustafa) [inline shading/coral-trino-parser/build.gradle:10]: Could you convert this to `compileOnly`?

Comment (wmoustafa) [reply shading/coral-trino-parser/build.gradle:10]: Also the `shadowJar` task needs to package `compileOnly`:

```
shadowJar {
  configurations = [
    project.configurations.compileOnly
  ]
  dependsOn(relocateShadowJar)
}
```

Comment (ljfgem) [reply shading/coral-trino-parser/build.gradle:10]: Any one change above will fail the local build of coral-trino:
```
> Task :coral-trino:compileJava FAILED
/Users/jiefli/code/coral-ljf-github/coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/PrestoParserDriver.java:8: error: package coral.shading.io.trino.sql.parser does not exist
import coral.shading.io.trino.sql.parser.ParsingOptions;
```

Comment (wmoustafa) [reply shading/coral-trino-parser/build.gradle:10]: You would need both.

Comment (ljfgem) [reply shading/coral-trino-parser/build.gradle:10]: Thanks, it could build, do I need to open a PR with that change?

Comment (wmoustafa) [reply shading/coral-trino-parser/build.gradle:10]: Yes, please (preferably after testing through local integration with Trino).

**Pull request merged**